File: go/etl/etl.go
package etl

import "strings"

// Transform converts old stuff to new stuff
func Transform(m map[int][]string) map[string]int {
	res := map[string]int{}
	for k, v := range m {
		for _, val := range v {
			res[strings.ToLower(val)] = k
		}
	}
	return res
}

File: csharp/etl/Etl.cs
using System;
using System.Collections.Generic;

public static class Etl
{
    public static Dictionary<string, int> Transform(Dictionary<int, string[]> old)
    {
        var res = new Dictionary<string, int>();

        foreach (KeyValuePair<int, string[]> entry in old) {
            foreach (string s in entry.Value) {
                res.Add(s.ToLower(), entry.Key);
            }
        }
        return res;
    }
}